influxdb_line_protocol/src/lib.rs

```diff
@@ -228,6 +231,30 @@ impl<'a> Display for ParsedLine<'a> {
     }
 }
 
+/// A `ParsedLine` that owns its data
+#[self_referencing]
+pub struct StaticParsedLine {
+    data: Arc<str>,
+
+    #[borrows(data)]
+    #[covariant]
+    line: ParsedLine<'this>,
```

Comment:
```suggestion
    /// `ParsedLine` backed by a structure that points to some part of `data`
    line: ParsedLine<'this>,
```

FWIW I wonder if having a Vec<ParsedLine<'this>> is what you will eventually want (so you can pass batches of parsed lines around without having to allocate for the `Arc` each time.

Reply:
I specifically need to be able to split per-line, into different IOx shards, and then subsequently within the producer